Scripts/DamageSource.cs

```diff
@@ -0,0 +1,76 @@
+﻿using BehaviorDesigner.Runtime;
+using BehaviorDesigner.Runtime.Tasks;
+using UnityEngine;
+
+namespace BehaviorDesigner
+{
+    [TaskCategory("Neo FPS")]
+    [TaskDescription("Do damage to a Neo FPS character.")]
+    [TaskName("Damage Source")]
+    public class DamageSource : Action, NeoFPS.IDamageSource
+    {
+        public float damageAmount = 5;
+        public SharedGameObject target;
+        public bool isCritical = false;
+        public float kickDistance = 0.02f;
+        public float kickRotation = 5f;
+        public float kickDuration = 0.5f;
+
+        #region Neo FPS IDamageSource
+        public NeoFPS.DamageFilter outDamageFilter
+        {
+            get { return NeoFPS.DamageFilter.AllDamageAllTeams; }
+            set { }
+        }
+
+        public NeoFPS.IController controller
+        {
+            get { return null; }
+        }
+
+        public Transform damageSourceTransform
+        {
+            get { return transform; }
+        }
+
+        public string description
+        {
+            get { return transform.name + "inflicts damage of " + damageAmount; }
```

Comment:
Description should just be a descriptive name (eg character name). It's up to the logger to format it in place